### PR TITLE
Add commandline-interface

### PR DESCRIPTION
**Pull request by SvantjeJung**
+ generate subcommands for server, client, single-player
  + ./battleship server [OPTIONS] <port> <name>
  + ./battleship client <ip> <port> <name>
  + ./battleship single <name>

+ server contains options for n-dimensional board and custom
ships-configuration

+ several TODOs that have to be implemented

### CODE REVIEW

**Comment by SvantjeJung** [comment]
Added a file-format to set a custom ships configuration - for later usage...

File: Cargo.lock [hunk 1]
```diff
@@ -1,4 +1,99 @@
 [root]
 name = "battleship"
 version = "0.1.0"
+dependencies = [
+ "clap 2.20.1 (registry+https://github.com/rust-lang/crates.io-index)",
+]
 
+[[package]]
```
Comment (Varaion) [inline Cargo.lock:8]: Hat es einen Grund, dass die .lock mitgepusht ist? :D

File: Cargo.toml [hunk 2]
```diff
@@ -2,11 +2,11 @@
 name = "battleship"
 version = "0.1.0"
 authors = [
-    "Dennis Altenhoff <daltenhoff@uos.de>",
-    "Tim Bohne <tbohne@uos.de>",
-    "Svantje Jung <svjung@uos.de>",
-    "Famke Lamberti <flamberti@uos.de>"
+    "Dennis Altenhoff <daltenhoff@uos.de> ",
```
Comment (Varaion) [inline Cargo.toml:5]: Are we missing a space here? :D

File: src/main.rs [hunk 3]
```diff
@@ -1,10 +1,154 @@
...
+            (about: "Server instance for the game")
+            (version: crate_version!())
+            (author: crate_authors!())
+            (@arg port: +required +takes_value "listening port")
+            (@arg name: +required +takes_value "Name of player")
+            (@arg size: -s --size +takes_value
+                "set N as board dimension => N x N [not yet implemented]"
+            )
+            (@arg ships: --ships +takes_value "load ship configuration [not yet implemented]")
+        )
```
Comment (Varaion) [inline src/main.rs]: Name of the placer for server?

File: src/main.rs [hunk 4]
```diff
@@ -1,10 +1,154 @@
...
+        (@subcommand client =>
+            (about: "Client instance for the game")
+            (version: crate_version!())
+            (author: crate_authors!())
+            (@arg ip: +required +takes_value "Connect to address")
+            (@arg port: +required +takes_value "Connect to port")
+            (@arg name: +required +takes_value "Name of player")
+        )
```
Comment (Varaion) [inline src/main.rs]: Name of the placer for client?

File: Cargo.lock (hunk 1 above)
Comment (SvantjeJung) [reply Cargo.lock:8]: Bei einem binary-project soll die `Cargo.lock` mit hochgeladen werden. http://doc.crates.io/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

File: src/main.rs [hunk 5]
```diff
@@ -1,10 +1,154 @@
...
     println!("Welcome to a round of 'battleship'");
+
+    match battleship.subcommand() {
+        ("server", Some(server_args)) => {
+            // required arguments
+            let port = validate_port(server_args.value_of("port").unwrap());
+            let name = server_args.value_of("name").unwrap();
+
+            // optional arguments
+            let mut size = 10;
```
Comment (Varaion) [inline src/main.rs]: Id rather go with a variable than using hard-coding :P

File: Cargo.lock (hunk 1 above)
Comment (Varaion) [reply Cargo.lock:8]: Alright, nvm then :)

File: Cargo.toml (hunk 2 above)
Comment (SvantjeJung) [reply Cargo.toml:5]: Where? I don't see anything missing. Spaces at the beginning of the other names for formatting purpose on runtime.

Comment (Varaion) [reply Cargo.toml:5]: Ah, I see 👍

File: src/main.rs (hunk 5 above)
Comment (SvantjeJung) [reply src/main.rs]: It's not hard-coded - just the default value if no custom value is given (see lines below)..

Comment (Varaion) [reply src/main.rs]: Doh!

**Pull request merged**